installer/scripts/service_control

```diff
@@ -16,137 +19,382 @@
+        WS_STATUS=$INTERNAL_ERROR
+        notification_exit $WS_STATUS
     fi
+
+    COUNTER=$(( $1 * 2 )) # Since we sleep 0.5 seconds, compute number of seconds
+    while [ $COUNTER -gt 0 ]; do
+        if ! this_omsagent_running; then
+            COUNTER=$(( $COUNTER - 1 ))
+            sleep 0.5
+        else
+            return 0
+        fi
+    done
+    if this_omsagent_running; then
+        return 0
+    fi
+
+    return 1 # Never came up.
 }
 
-wait_until_omsagent_stops()
+wait_until_omsagent_stopped()
 {
     # Required parameter: Number of seconds to wait for agent to stop
     if [ -z "$1" -o "$1" -le 0 ]; then
-        echo "Function \"wait_until_omsagent_stops\" called with invalid parameter"
-        exit 1
+        echo "INTERNAL ERROR:  Function \"wait_until_omsagent_stops\" called with invalid parameter"
+        notification_exit $INTERNAL_ERROR
     fi
 
     COUNTER=$(( $1 * 2 )) # Since we sleep 0.5 seconds, compute number of seconds
     while [ $COUNTER -gt 0 ]; do
-        is_omsagent_running && return $?
-        COUNTER=$(( $COUNTER - 1 ))
-        sleep 0.5
+        if this_omsagent_running; then
+            COUNTER=$(( $COUNTER - 1 ))
+            sleep 0.5
+        else
+            remove_PIDfile_unless_omsagent_running
+            return $? # Should only be non-zero if a PIDFILE remains.
+        fi
     done
+    if ! this_omsagent_running; then
+        remove_PIDfile_unless_omsagent_running
+        return 0
+    fi
 
-    # One final try for accurate return status (just return status from the call)
-    is_omsagent_running
+    return 1 # Never stopped.
 }
 
 stop_omsagent_process()
 {
-    # First try sending a graceful shutdown request.
-    # If omsagent doesn't stop, forcefully kill it.
-    #
-    # TODO: omsagent doesn't currently gracefully shut down, we're investigating this later
-
-    is_omsagent_running && return $?
-
-    # kill -sigterm `cat $PIDFILE`
-    # wait_until_omsagent_stops 10 && return $?
+    remove_PIDfile_unless_omsagent_running
+    if ! this_omsagent_running; then
+        # It is stopped.  Presume okay.
+        return 0
+    fi
+    WS_STATUS=0 # Try yet again.
 
     echo -n "(Forced) "
     kill -sigkill `cat $PIDFILE`
-    wait_until_omsagent_stops 5
-    return $?
+    if [ $? -eq 0 ]; then
+        wait_until_omsagent_stopped $STOP_PROCESS_WAIT_MAX
+    fi
+    WS_STATUS=$?
+    return $WS_STATUS
+}
+
+generateProcedureName()
+{
+    case "$1" in
+        disable)    echo "disable_omsagent_service";;
+        enable)     echo "enable_omsagent_service";;
+        reload)     echo "restart_omsagent";;
+        restart)    echo "restart_omsagent";;
+        start)      echo "start_omsagent";;
+        stop)       echo "stop_omsagent";;
+        *)          echo "INTERNAL ERROR:  '$1' is not a valid procedural activity."
+                    notification_exit INTERNAL_ERROR;;
+    esac
 }
 
-start_omsagent_process()
+# ## Iterators #### 
+
+check_non_oms_and_invoke()
```

Comment:
Indentation in this method is still inconsistent - please fix these methods: check_non_oms_and_invoke, check_oms_and_invoke, service_template, and the main procedure case statement!